instrumentation/netty/netty-4.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/Netty41ClientSslTest.java

```diff
@@ -0,0 +1,260 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class Netty41ClientSslTest {
+
+  @RegisterExtension
+  static InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private HttpClientTestServer server;
+
+  private EventLoopGroup eventLoopGroup;
+
+  @SuppressWarnings("UnusedMethod")
+  private static Bootstrap createBootstrap(
+      EventLoopGroup eventLoopGroup, List<String> enabledProtocols) {
+    Bootstrap bootstrap = new Bootstrap();
+    bootstrap
+        .group(eventLoopGroup)
+        .channel(NioSocketChannel.class)
+        .handler(
+            new ChannelInitializer<SocketChannel>() {
+              @Override
+              protected void initChannel(SocketChannel socketChannel) throws Exception {
+                ChannelPipeline pipeline = socketChannel.pipeline();
+
+                SslContext sslContext = SslContextBuilder.forClient().build();
+                SSLEngine sslEngine = sslContext.newEngine(socketChannel.alloc());
+                if (enabledProtocols != null) {
+                  sslEngine.setEnabledProtocols(enabledProtocols.toArray(new String[0]));
+                }
+                pipeline.addLast(new SslHandler(sslEngine));
+
+                pipeline.addLast(new HttpClientCodec());
+              }
+            });
+    return bootstrap;
+  }
+
+  @BeforeEach
+  void setUp() {
+    server = new HttpClientTestServer(testing.getOpenTelemetry());
+    server.start();
+    eventLoopGroup = new NioEventLoopGroup();
+    System.setProperty("otel.instrumentation.netty.connection-telemetry.enabled", "true");
+    System.setProperty("otel.instrumentation.netty.ssl-telemetry.enabled", "true");
+  }
+
+  @AfterEach
+  void tearDown() throws ExecutionException, InterruptedException, TimeoutException {
+    server.stop().get(10, TimeUnit.SECONDS);
+    eventLoopGroup.shutdownGracefully();
+    System.clearProperty("otel.instrumentation.netty.connection-telemetry.enabled");
+    System.clearProperty("otel.instrumentation.netty.ssl-telemetry.enabled");
+  }
+
+  @Test
+  @DisplayName("should fail SSL handshake")
+  @SuppressWarnings("InterruptedExceptionSwallowed")
+  public void testFailSslHandshake() throws Exception {
+    Bootstrap bootstrap = createBootstrap(eventLoopGroup, Collections.singletonList("SSLv3"));
+    URI uri = server.resolveHttpsAddress("/success");
+    DefaultFullHttpRequest request =
+        new DefaultFullHttpRequest(
+            HttpVersion.HTTP_1_1, HttpMethod.GET, uri.getPath(), Unpooled.EMPTY_BUFFER);
+    request.headers().set(HttpHeaderNames.HOST, uri.getHost() + ":" + uri.getPort());
+
+    Throwable thrownException = null;
+    try {
+      testing.runWithSpan(
+          "parent",
+          () -> {
+            Channel channel = bootstrap.connect(uri.getHost(), uri.getPort()).sync().channel();
```

Comment:
groovy test closes the channel as part of the cleanup at the end of the test. Search for `AutoCleanupExtension` in instrumentation tests, you can use it to close the channel at the end of the test.

Reply:
done